cpp/dolfinx/fem/FiniteElement.h

```diff
@@ -417,8 +425,7 @@ class FiniteElement
         const std::function<void(std::span<U>, std::span<const std::uint32_t>,
                                  std::int32_t, int)>
             sub_function
-            = _sub_elements[0]->template dof_transformation_function<U>(
-                ttype);
+            = _sub_elements[0]->template dof_transformation_fn<U>(ttype);
```

Comment:
@mscroggs This looks fishy (before and after the change). These lines are the only 'left-apply' functions in a function that is otherwise all 'right-apply'. What do you think it should be?

Reply:
Most likely this:

```suggestion
            = _sub_elements[0]->template dof_transformation_right_fn<U>(ttype);
```

Reply:
Hmm, with this change:

https://github.com/FEniCS/dolfinx/compare/garth/transform-name-updates...garth/transform-test

tests fail:

https://github.com/FEniCS/dolfinx/actions/runs/8764139051

Any ideas?

Reply:
I'll take a closer look on Monday. I guess there's a good reason for not having the right that I need to re-work out and add a comment explaining

Reply:
There are some more transforms that look fishy and there be some cancellation by two incorrect transforms - I'll make a separate PR to highlight.